extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -1,32 +1,38 @@
-import {Layout_Enum, applyFillContent} from '#core/dom/layout';
+import * as Preact from '#core/dom/jsx';
+import {Layout_Enum} from '#core/dom/layout';
 
-import {AmpStoryPageAttachment} from 'extensions/amp-story/1.0/amp-story-page-attachment';
-
-const TAG = 'amp-story-shopping-attachment';
-
-export class AmpStoryShoppingAttachment extends AmpStoryPageAttachment {
+export class AmpStoryShoppingAttachment extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
 
-    /** @private {string} */
-    this.myText_ = TAG;
-
-    /** @private {?Element} */
-    this.container_ = null;
+    /** @private @return {!Element} */
+    this.attachmentEl_ = null;
   }
 
   /** @override */
   buildCallback() {
-    super.buildCallback();
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    applyFillContent(this.container_, /* replacedContent */ true);
+    this.attachmentEl_ = (
+      <amp-story-page-attachment
+        layout="nodisplay"
+        theme={this.element.getAttribute('theme')}
+      ></amp-story-page-attachment>
+    );
+    this.element.appendChild(this.attachmentEl_);
+  }
+
+  /**
+   * @param {boolean=} shouldAnimate
+   * @return {Promise}
```

Comment:
This should be updated to `@return {!Promise}` because all non-primitive types should be annotated with `!` or `?` (primitives being `string`, `number`, `boolean`, `undefined`, `null`). In this case, we know from the `@return` of `getImpl()` that it returns a non-null Promise, so `{!Promise}` is appropriate here.

Reply:
TIL. Thank you for explaining ! and ? :)